src/UpdateIProductList.ts

```diff
@@ -0,0 +1,28 @@
+import { loadPromotions } from "./Dependencies";
+import { IProduct } from "./IProduct";
+
+export function UpdateIProductList(products: IProduct[]): IProduct[] {
```

Comment:
the name of this function better reveal the invention, in this scenario it apply the promotion info